godot-core/src/docs.rs

```diff
@@ -34,23 +34,33 @@ pub struct StructDocs {
 /// ```
 #[derive(Clone, Copy, Debug, Default)]
 pub struct InherentImplDocs {
-    pub methods: &'static str,
-    pub signals: &'static str,
-    pub constants: &'static str,
+    pub methods: Option<&'static str>,
+    pub signals: Option<&'static str>,
+    pub constants: Option<&'static str>,
 }
 
 #[derive(Default)]
 struct DocPieces {
     definition: StructDocs,
     inherent: InherentImplDocs,
-    virtual_methods: &'static str,
+    virtual_methods: Option<&'static str>,
+}
+
+#[doc(hidden)]
+/// Generates documentation for given block if there is anything to document
+/// Returns empty string otherwise
+fn block_docstring_or_empty(block: Option<&'static str>, tag: &'static str) -> String {
+    match block {
+        Some(s) => format!("<{tag}>{s}</{tag}>"),
+        _ => "".to_string(),
+    }
 }
```

Comment:
Is there any chance you could move the tags into the block at macro time instead?
```suggestion
fn block_docstring_or_empty(block: Option<&'static str>, tag: &'static str) -> String {
    block.map(|s| format!("<{tag}>{s}</{tag}>")).unwrap_or_default()
}
```

Reply:
It is an option for signals & constants – methods&virtual methods needs to be formatted during runtime :thinking: